src/explorer/connectionTreeItem.ts

```diff
@@ -67,21 +69,24 @@ export default class ConnectionTreeItem
     childrenCache: {
       [key: string]: DatabaseTreeItem | StreamProcessorTreeItem;
     }; // Existing cache.
+    source: ConnectionSource;
   }) {
     super(
       connectionController.getSavedConnectionName(connectionId),
       collapsibleState
     );
 
-    if (
+    const isConnected =
       connectionController.getActiveConnectionId() === connectionId &&
       !connectionController.isDisconnecting() &&
-      !connectionController.isConnecting()
-    ) {
-      this.contextValue = ConnectionItemContextValues.connected;
-    }
+      !connectionController.isConnecting();
+
+    this.contextValue = `${isConnected ? 'connected' : 'disconnected'}${
+      source === 'user' ? '' : 'Preset'
+    }ConnectionTreeItem`;
 
     this.connectionId = connectionId;
+    this.source = source;
     this._connectionController = connectionController;
     this.isExpanded = isExpanded;
     this._childrenCache = childrenCache;
```

Comment:
Can't comment on the relevant line, but you can be a good [boy scout](https://biratkirat.medium.com/step-8-the-boy-scout-rule-robert-c-martin-uncle-bob-9ac839778385) and add some returns to the functions on line 212 and 234 to fix those warnings.